tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -521,6 +521,13 @@ def fit(self,
                 data. e.g. 500 hand-labeled samples (perhaps a hard human task), whole data set (unlabeled) is thousands/millions.
                 However, this isn't the only use case. Given enough unlabeled data(millions of rows), you may see improvements
                 to any amount of labeled data.
+            weight_column : str, default = None
```

Comment:
I'm in favor of renaming to sample_weight to align with other packages. I think the fact we use a str instead of the actual values isn't too big of a deal.

Reply:
Also, I think we should support `sample_weight='auto'`, where we automatically set weights to whatever we think will get the best score for the eval_metric.

Reply:
renamed, and added TODO to support 'auto' in the future